feature-libs/product/future-stock/core/services/future-stock.service.spec.ts

```diff
@@ -9,14 +14,20 @@ class MockFutureStockConnector implements Partial<FutureStockConnector> {
   getFutureStock = createSpy().and.callFake(() => of(mockFutureStocks));
 }
 
-const mockUserId = 'current';
+const mockUserId = OCC_USER_ID_CURRENT;
 
 class MockUserIdService implements Partial<UserIdService> {
   takeUserId(): Observable<string> {
     return of(mockUserId);
   }
 }
 
+class MockUserIdAnonymousService implements Partial<UserIdService> {
```

Comment:
totally fine to me as it's a way of testing

----

I would have preferred making use of the createSpy().

What would you do to make things work instead of configuring the TestBed on demand would be the following:

1 - keep the TestBed module configuration in the before each as it was before

2 - make use of the spy for the `takeUserId`

```
class MockUserIdService implements Partial<UserIdService> {
  takeUserId = createSpy().and.returnValue(of(mockUserId));
}
```

3 - inject the service in question

```
userIdService = TestBed.inject(UserIdService)
```

4 - in the describe/context in question, you would change the return value as follows

```
userIdService.takeUserId = createSpy().and.returnValue(of(theMockUserInQuestion));
```

Reply:
I tried this method using spies but it would not take effect. I am guessing it is because of the usage of `takeUserId()` within the pipe that doesn't allow it to be "called" per se and therefore the spy will not return its value.

Reply:
> 

future references as I did what I mentioned above on this recent merged commit https://github.com/SAP/spartacus/commit/f6b176dbe06d6677750e4cbd23c6462c3e023972#diff-a3a961dff01869c8cea9fc6650d0e7df092943420402eacee9eb816017f0b041R106

Reply:
> I tried this method using spies but it would not take effect. I am guessing it is because of the usage of `takeUserId()` within the pipe that doesn't allow it to be "called" per se and therefore the spy will not return its value.

future references as I did what I mentioned above on this recent merged commit https://github.com/SAP/spartacus/commit/f6b176dbe06d6677750e4cbd23c6462c3e023972#diff-a3a961dff01869c8cea9fc6650d0e7df092943420402eacee9eb816017f0b041R106